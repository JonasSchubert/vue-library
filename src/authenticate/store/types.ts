export const ActionTypes = {
  login: 'login',
  logout: 'logout',
  validateToken: 'validate-token'
};

export const GetterTypes = {
  error: 'error',
  isAdministrator: 'isAdministrator',
  isLoading: 'isLoading',
  isUser: 'isUser',
  loggedIn: 'loggedIn',
  token: 'token',
  userRoles: 'userRoles'
};

export const ModuleType = 'authenticate';

export const MutationTypes = {
  setError: 'setError',
  setIsLoading: 'setIsLoading',
  setToken: 'setToken',
  setUserRoles: 'setUserRoles'
};

export const RouteTypes = {
  authenticate: 'authenticate'
};
