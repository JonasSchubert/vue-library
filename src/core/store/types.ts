export const GetterTypes = {
  isAnythingLoading: 'isAnythingLoading'
};

export const MutationTypes = {
  decreaseRequests: 'decreaseRequests',
  increaseRequests: 'increaseRequests'
};
