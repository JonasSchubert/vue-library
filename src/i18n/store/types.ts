export const ActionTypes = {
  getAllTranslations: 'getAllTranslations',
  getAvailableLocales: 'getAvailableLocales',
  getLocaleTranslations: 'getLocaleTranslations'
};

export const GetterTypes = {
  availableLocales: 'availableLocales',
  currentLocale: 'currentLocale',
  error: 'error',
  locales: 'locales'
};

export const MutationTypes = {
  setAvailableLocales: 'setAvailableLocales',
  setCurrentLocale: 'setCurrentLocale',
  setError: 'setError',
  setLocales: 'setLocales',
  updateLocales: 'updateLocales'
};

export const RouteTypes = {
  getAllTranslations: 'i18n/GetAllIetfTranslations',
  getAvailableLocales: 'i18n/GetAvailableIetf',
  getLocaleTranslations: 'i18n/GetIetfTranslations'
};
