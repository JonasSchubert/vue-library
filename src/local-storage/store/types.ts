export const ActionTypes = {
  loadHistory: 'loadHistory',
  loadSettings: 'loadSettings',
  updateIsDarkTheme: 'updateIsDarkTheme',
  updateLastRoutes: 'updateLastRoutes'
};

export const GetterTypes = {
  history: 'history',
  isDarkTheme: 'isDarkTheme',
  lastRoutes: 'lastRoutes',
  settings: 'settings'
};

export const LocalStorageTypes = {
  history: 'i18n-service-history',
  settings: 'i18n-service-settings'
};

export const ModuleType = 'local-storage';

export const MutationTypes = {
  setHistory: 'setHistory',
  setSettings: 'setSettings',
  updateIsDarkTheme: 'updateIsDarkTheme',
  updateLastRoutes: 'updateLastRoutes'
};
