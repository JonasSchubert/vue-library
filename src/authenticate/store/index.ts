import { Module } from 'vuex';
import { AuthenticateModuleSetup, AuthenticateState, LoginResponse } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createAuthenticateState } from './state';

export const createAuthenticateModule = <TRootState, T extends LoginResponse>({ cookieKeyAuthenticationToken, customGetters, daysTilExpiredAuthenticationCookie }: AuthenticateModuleSetup<TRootState, T>): Module<AuthenticateState<T>, TRootState> => ({
  namespaced: true,
  actions: createActions(),
  getters: createGetters(customGetters),
  mutations: createMutations({ cookieKeyAuthenticationToken, daysTilExpiredAuthenticationCookie }),
  state: createAuthenticateState()
});
