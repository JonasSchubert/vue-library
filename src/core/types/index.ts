export * from './ga-colors.type';
