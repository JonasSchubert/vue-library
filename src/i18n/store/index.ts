import { Module } from 'vuex';
import { I18nModuleConfig, I18nState } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createState } from './state';

export const createModule = <T>(config: I18nModuleConfig): Module<I18nState, T> => ({
  namespaced: true,
  actions: createActions<T>(config),
  getters: createGetters<T>(),
  mutations: createMutations(config),
  state: createState(config)
});
