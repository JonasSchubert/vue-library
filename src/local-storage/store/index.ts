import { Module } from 'vuex';
import { LocalStorageState } from '../models';
import { createActions } from './actions';
import { createGetters } from './getters';
import { createMutations } from './mutations';
import { createLocalStorageState } from './state';

export const createLocalStorageModule = <T>(): Module<LocalStorageState, T> => ({
  namespaced: true,
  actions: createActions<T>(),
  getters: createGetters<T>(),
  mutations: createMutations(),
  state: createLocalStorageState()
});
