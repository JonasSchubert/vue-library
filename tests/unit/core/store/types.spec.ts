import { GetterTypes, MutationTypes } from '@/core/store/types';

describe('GetterTypes', () => {
  test('should not change unexpected', () => {
    // Assert
    expect(GetterTypes).toMatchSnapshot();
  });
});

describe('MutationTypes', () => {
  test('should not change unexpected', () => {
    // Assert
    expect(MutationTypes).toMatchSnapshot();
  });
});
