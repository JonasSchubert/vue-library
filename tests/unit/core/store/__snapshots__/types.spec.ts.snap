// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GetterTypes should not change unexpected 1`] = `
Object {
  "isAnythingLoading": "isAnythingLoading",
}
`;

exports[`MutationTypes should not change unexpected 1`] = `
Object {
  "decreaseRequests": "decreaseRequests",
  "increaseRequests": "increaseRequests",
}
`;
